gcloud/storage/acl.py

```diff
@@ -351,23 +349,24 @@ def get_entities(self):
         self._ensure_loaded()
         return list(self.entities.values())
 
-    @staticmethod
-    def _client_or_connection(client):
-        """Temporary method to get a connection from a client.
+    @property
+    def client(self):
+        """Abstract getter for the object client."""
+        raise NotImplementedError
 
-        If the client is null, gets the connection from the environment.
+    def _require_client(self, client):
+        """Check client or verify over-ride.
 
         :type client: :class:`gcloud.storage.client.Client` or ``NoneType``
-        :param client: Optional. The client to use.  If not passed, falls back
-                       to default connection.
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current object.
 
-        :rtype: :class:`gcloud.storage.connection.Connection`
-        :returns: The connection determined from the ``client`` or environment.
+        :rtype: :class:`gcloud.storage.client.Client`
+        :returns: The client passed in or the currently bound client.
         """
         if client is None:
-            return _require_connection()
-        else:
-            return client.connection
+            client = self.client
+        return client
```

Comment:
I agree. Wasn't sure what to do about it, didn't want to make the classes too close cousins just yet. Shall I file a bug?

Also, `ACL` still needs a lot of TLC (actual design) from us. I think you'll need to tackle it as part of BigQuery too.